### DISCUSSION TITLE
Can't figure out how to use /v1/initwallet byte macaroon.

### DISCUSSION THREAD

**Discussion by lucas-naman** [lightningnetwork/lnd#5934] [Q&A] [answered]
Hello,

I'm currently trying to make my first project with LND.

In order to do so I need to initialize my wallet using the rest LND api endpoint: /v1/initwallet.
This actually works fine, but what I can't figure out is: how to use the "admin_macaroon" I get inside the response of the request.
I want to use it (the macaroon) in the headers when I call the endpoint "/v1/getinfo" for example.

Can anyone help me with that ?

If it's my question is not clear do not hesitate to ask for precisions.

Thanks

**Comment by guggero** [comment, marked as answer]
You don't _have_ to use the macaroon that is returned to you. You can also use one of the files created by default in `~/.lnd/data/chain/bitcoin/<network>/*.macaroon`.
In any case, the macaroon must be hex encoded in the header field:

```
  headers: {
    'Grpc-Metadata-macaroon': macaroon,
  },
```

If you use the one you get in the response to `/v1/initwallet` you might need to convert it, as it's base64 in REST responses: `base64 -> raw bytes/buffer -> encode as hex`.
If you use a file, you get the raw bytes from the file and also need to hex encode them.

**Reply by lucas-naman** [reply]
Thanks for the fast answer, I wrote my reply to you as an other answer.

**Comment by lucas-naman** [comment]
Alright thanks.

What I was missing was the fact that the REST responded a `base64` string in the `"admin_macaroon"` field.
In any case, if someone has the same problem, here is what I did in python3 to get a string usable in the header for future requests:

`import base64`
`byte_macaroon = base64.b64decode(base64_macaroon)`
`hex_macaroon = byte_macaroon.hex()`

`headers: {`
    `'Grpc-Metadata-macaroon': hex_macaroon,`
` }`
